include/cocaine/trace/trace.hpp

```diff
@@ -174,29 +140,72 @@ class trace_t {
     mem_fn(Method m) -> callable_wrapper_t<decltype(std::mem_fn(std::forward<Method>(m)))>;
 
 private:
+
     static
     uint64_t
-    generate_id() {
-        static std::random_device rd;
-        static std::mt19937 gen(rd());
-        // Stupid zipkin-web can not handle unsigned ids. So we limit to signed diapason.
-        static std::uniform_int_distribution<uint64_t> dis(1, std::numeric_limits<uint64_t>::max()/2-1);
-        return dis(gen);
-    }
+    generate_id();
+
+    static
```

Comment:
I'd rather make it a free function, because it hasn't much to do with `trace` scope.